### ISSUE TITLE
Set custom logger before startKoin() in Android

### ISSUE THREAD

**Issue by dkrivoruchko** [Ekito/koin#50]
Is there any way to set custom Logger for koin before startKoin in android App?
Or maybe set Logger as a parameter for startKoin.
Main issue with current AndroidLogger() is that it's set in Application.startKoin() and make visible all KOIN logs to everyone in release  builds.

**Comment by arnaudgiuliani** [comment]
Hello,

Koin logging is set by default with startKoin. I will add the ability to set the needed logger.

Stay in touch.

**Comment by dkrivoruchko** [comment]
Thanks

**Comment by erickok** [comment]
Agreed, it should never log anything by default.

**Comment by fredy-mederos** [comment]
My solution to this issue was to manually remove the Android logger.
You can use it meanwhile...
```kotlin
//startKoin here
 Koin.logger = if (isDebugMode())
            org.koin.android.logger.AndroidLogger() else
            org.koin.log.EmptyLogger()
```
Passing an extra param to enable logs are the best solution.

**Comment by arnaudgiuliani** [comment]
I was thinking about putting `Koin.logger` as nullable and check in startKoin if it's not already set, to put the default value. It would allow to write `Koin.logger = EmptyLogger()` before startKoin, if we are in release.

**Comment by dkrivoruchko** [comment]
I'll prefer an optional parameter `Logger` for Application.startKoin()

**Comment by arnaudgiuliani** [comment]
Ok to make it as an optional param.

**Comment by arnaudgiuliani** [comment]
Available in `0.9.0-alpha-5`. startKoin has now an optional parameter for Koin logger `logger`.
